### PR TITLE
fix: MDP-3305 store imageInfo in db pushed to NTB

### PR DESCRIPTION
**Pull request by sheikhimtiaz-cefalo**

### CODE REVIEW

File: src/main/java/no/mentordigital/arctontbuploader/model/ImageMigrationStatus.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+@Table(
+    name = "imageMigrationStatus",
+    indexes = {@Index(columnList = "id", unique = true)})
+@Builder
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@DynamicInsert
+@DynamicUpdate
```
Comment (sultanmahmud0) [inline src/main/java/no/mentordigital/arctontbuploader/model/ImageMigrationStatus.java:19]: I am not sure why we need all these annotations, only 
@Entity
@Table
@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor should be enough.

File: src/main/java/no/mentordigital/arctontbuploader/service/ImageMigrationStatusService.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+@Service
+@RequiredArgsConstructor
+public class ImageMigrationStatusService {
+  private final ImageMigrationStatusRepository imageMigrationStatusRepository;
+
+  public void saveImageMigrationStatus(ImageMigrationStatus imageMigrationStatus) {
+    imageMigrationStatusRepository.save(imageMigrationStatus);
+  }
+
+  public ImageMigrationStatus fetchImageMigrationStatus(String uniqueId) {
```
Comment (sultanmahmud0) [inline src/main/java/no/mentordigital/arctontbuploader/service/ImageMigrationStatusService.java:17]: Please rename the parameter name to "ArcId" for better understanding.

File: src/main/java/no/mentordigital/arctontbuploader/utils/MigrationStatusUtil.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package no.mentordigital.arctontbuploader.utils;
+
+import no.mentordigital.arctontbuploader.model.ImageMigrationStatus;
+import no.mentordigital.arctontbuploader.service.ImageMigrationStatusService;
+import no.mentordigital.arctontbuploader.webclient.arc.model.ImageInStory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
```
Comment (sultanmahmud0) [inline src/main/java/no/mentordigital/arctontbuploader/utils/MigrationStatusUtil.java:9]: For an utility class it is better to use "@Component".

File: src/main/java/no/mentordigital/arctontbuploader/utils/MigrationStatusUtil.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+public class MigrationStatusUtil {
+  @Autowired ImageMigrationStatusService imageMigrationStatusService;
+
+  public boolean isMigrated(ImageInStory imageInStory) {
```
Comment (sultanmahmud0) [inline src/main/java/no/mentordigital/arctontbuploader/utils/MigrationStatusUtil.java:13]: I would rather say we should create a method as "shouldMigrate()" as this
`    ImageMigrationStatus imageMigrationStatus =
        imageMigrationStatusService.fetchImageMigrationStatus(imageInStory.getImageId());

if (imageMigrationStatus == null || !imageMigrationStatus.isSuccess()) return true;

else  imageInStory.getLastUpdatedDate().after(imageMigrationStatus.getLast_updated_time());

       

`

File: src/main/java/no/mentordigital/arctontbuploader/webclient/arc/ArcReaderClient.java [hunk 5]
```diff
@@ -64,6 +67,101 @@ public void scanAndHandle(ScanQueryModel scanQuery) {
             () -> log.info("No more results to scan in this batch"));
   }
 
+  public void scanPhotosAndHandle(ScanQueryModel scanQuery, String apiPath) {
+    String batchid = "arc-import-test"; // ; + String.valueOf(Instant.now().getEpochSecond());
```
Comment (sultanmahmud0) [inline src/main/java/no/mentordigital/arctontbuploader/webclient/arc/ArcReaderClient.java:71]: Ok for now but to test we should have a unique batch number.

File: src/main/java/no/mentordigital/arctontbuploader/webclient/arc/ArcReaderClient.java [hunk 6]
```diff
@@ -64,6 +67,101 @@ public void scanAndHandle(ScanQueryModel scanQuery) {
...
+
+    var uriSpec =
+        client
+            .get()
+            .uri(
+                uribuilder ->
+                    uribuilder
+                        .path("/photos")
+                        .queryParam("q", "primaryWebsite:{website}")
+                        .queryParam("limit", "10")
```
Comment (sultanmahmud0) [inline src/main/java/no/mentordigital/arctontbuploader/webclient/arc/ArcReaderClient.java:153]: Are we keeping the limit to 10 ? Isn't that a bit small as we need to export a large number of images ?

File: src/main/java/no/mentordigital/arctontbuploader/webclient/arc/ArcReaderClient.java [hunk 7]
```diff
@@ -64,6 +67,101 @@ public void scanAndHandle(ScanQueryModel scanQuery) {
...
+                        .queryParam("offset", offset)
+                        .queryParam(
+                            "distributorId",
+                            appConfig
+                                .getDistributor()
+                                .getOrDefault(
+                                    scanQuery.getWebsite(), "27ca5f68-c054-48ba-a09c-b075e158a488"))
```
Comment (sultanmahmud0) [inline src/main/java/no/mentordigital/arctontbuploader/webclient/arc/ArcReaderClient.java:160]: We need to pass the distributor value dynamically.

File: src/main/resources/application.yml [hunk 8]
```diff
@@ -16,7 +16,24 @@ arc-api:
...
 
 
+spring:
+  jackson:
+    mapper:
+      ACCEPT_CASE_INSENSITIVE_ENUMS: true
+  datasource:
+    url: ${dburl:jdbc:mysql://localhost:3306/usermanagement}
+    username: root
+    password: Iamnumber61$
```
Comment (sultanmahmud0) [inline src/main/resources/application.yml:26]: This should be removed from actual application YAML and h2 database should be used.

File: src/main/java/no/mentordigital/arctontbuploader/model/ImageMigrationStatus.java (hunk 1 above)
Comment (sheikhimtiaz-cefalo) [reply src/main/java/no/mentordigital/arctontbuploader/model/ImageMigrationStatus.java:19]: Right. removed the last two unused annotations. 

Fixed it.

File: src/main/java/no/mentordigital/arctontbuploader/service/ImageMigrationStatusService.java (hunk 2 above)
Comment (sheikhimtiaz-cefalo) [reply src/main/java/no/mentordigital/arctontbuploader/service/ImageMigrationStatusService.java:17]: FIxed it.

File: src/main/java/no/mentordigital/arctontbuploader/utils/MigrationStatusUtil.java (hunk 4 above)
Comment (sheikhimtiaz-cefalo) [reply src/main/java/no/mentordigital/arctontbuploader/utils/MigrationStatusUtil.java:13]: Fixed it.

File: src/main/java/no/mentordigital/arctontbuploader/utils/MigrationStatusUtil.java (hunk 3 above)
Comment (sheikhimtiaz-cefalo) [reply src/main/java/no/mentordigital/arctontbuploader/utils/MigrationStatusUtil.java:9]: Fixed it.

File: src/main/java/no/mentordigital/arctontbuploader/webclient/arc/ArcReaderClient.java (hunk 6 above)
Comment (sheikhimtiaz-cefalo) [reply src/main/java/no/mentordigital/arctontbuploader/webclient/arc/ArcReaderClient.java:153]: Fixed it.

File: src/main/java/no/mentordigital/arctontbuploader/webclient/arc/ArcReaderClient.java (hunk 7 above)
Comment (sheikhimtiaz-cefalo) [reply src/main/java/no/mentordigital/arctontbuploader/webclient/arc/ArcReaderClient.java:160]: Fixed it.

File: src/main/resources/application.yml (hunk 8 above)
Comment (sheikhimtiaz-cefalo) [reply src/main/resources/application.yml:26]: Fixed it.

**Review by sultanmahmud0 — APPROVED** [review]

**Pull request merged**